### ISSUE TITLE
Stuck Process

### ISSUE THREAD

**Issue by Rucknar** [rancher/rancher#3506]
```
Component	Version
Rancher	v0.56.1
Cattle	v0.141.1
User Interface	v0.86.0
Rancher Compose	v0.7.1
```
Following on from a failure over a month ago, i have the following issue:
* There is an additional network agent on one host listed in the UI, with a status of `In Progress`. It doesn't actually exist on the host and the other network agent is running fine.
* I have a stuck process under the admin/processes tab for  `instance.restore`, when i try drill down into this task i get a blank page. 
* The following message is logged out once a minute to the rancher-server instance:
```
2016-02-02 18:07:15,928 ERROR [:] [] [] [] [utorService-222] [i.c.p.e.e.i.ProcessEventListenerImpl] Unknown exception running process [instance.restore:209606] on [754] java.lang.IllegalStateException: Attempt to cancel when process is still transitioning
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.runDelegateLoop(DefaultProcessInstanceImpl.java:189) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.executeWithProcessInstanceLock(DefaultProcessInstanceImpl.java:156) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$1.doWithLock(DefaultProcessInstanceImpl.java:106) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl$1.doWithLock(DefaultProcessInstanceImpl.java:103) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.process.impl.DefaultProcessInstanceImpl.execute(DefaultProcessInstanceImpl.java:103) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.engine.eventing.impl.ProcessEventListenerImpl.processExecute(ProcessEventListenerImpl.java:66) ~[cattle-framework-engine-0.5.0-SNAPSHOT.jar:na]
        at sun.reflect.GeneratedMethodAccessor838.invoke(Unknown Source) ~[na:na]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_91]
        at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_91]
        at io.cattle.platform.eventing.annotation.MethodInvokingListener$1.doWithLockNoResult(MethodInvokingListener.java:76) [cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:7) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:3) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) [cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.eventing.annotation.MethodInvokingListener.onEvent(MethodInvokingListener.java:72) [cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
        at io.cattle.platform.eventing.impl.AbstractThreadPoolingEventService$2.doRun(AbstractThreadPoolingEventService.java:135) [cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.NoExceptionRunnable.runInContext(NoExceptionRunnable.java:13) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:49) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:108) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:46) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_91]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_91]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_91]
```

It came about when i had issues with an older version of rancher that are now resolved with 0.56.1, just interested in understanding how i can clear down this erroneous job from the server.

**Comment by deniseschannon** [comment]
Can you find the instance in the API and do a delete from the API?

**Comment by Rucknar** [comment]
Sadly not:
```
HTTP/1.1 500
Date: Thu, 04 Feb 2016 09:17:11 GMT
X-Api-Client-Ip: 10.40.16.10
Server: Jetty(9.2.11.v20150529)
X-Api-Account-Id: 1a8
Content-Type: application/json; charset=utf-8
X-Api-Schemas: http://X.X.X.X:8080/v1/projects/1a8/schemas
Content-Length: 176
Expires: Thu, 01 Jan 1970 00:00:00 GMT

{
"id": "7cde9e5e-677d-459e-ace9-cef1c4e04b1c",
"type": "error",
"links": { },
"actions": { },
"status": 500,
"code": "Internal Server Error",
"message": "Internal Server Error",
"detail": null,
}
```

**Comment by dominikhahn** [comment]
+1

**Comment by Rucknar** [comment]
Might i be able to remove this ghost entity from the database? If there are any pointers i'm happy to give it a try.

**Comment by Rucknar** [comment]
Rightly or wrongly i've managed to remove this instance. Below are the steps i used to do so, i'd only run this after taking a backup of your DB first and accepting that we sort of 'took a punt' on this one, all of the below is done using the ID's we had, use the one from the stuck process. 

First we grabbed the instance_id from the running process's tab, in this case it was `1pi209606`
validate this using the following command: `select * from instance WHERE instance_id="1pi209606"`
This should return a row, in our case running for 2 months. 

We then ran the following commands, these were done first as they had a foreign key relationship with the `instance` table:
```
delete from instance_host_map WHERE instance_id="1pi209606";
delete from instance_label_map WHERE instance_id="1pi209606"
delete from nic WHERE instance_id="1pi209606"
delete from ip_address_nic_map where id="1";
delete  from ip_address_nic_map WHERE nic_id="1";
delete from nic WHERE instance_id="1pi209606"
delete from port where instance_id="1pi209606";
delete from service_expose_map  WHERE id="1pi209606";
delete from volume  WHERE id="1pi209606";
```
Then finally we removed it from the instance table:
```
delete  from instance WHERE id="1pi209606";
```
FInally, if you click through on that stuck process, grab the `	Resource Type:Id` value, then run:
```
delete from process_instance where resource_id="754";
```
We then restarted our rancher server to ensure it was all gone, we no longer see errors in the logs or see the stuck job.
